nautobot/core/api/schema.py

```diff
@@ -36,6 +37,49 @@ class NautobotAutoSchema(AutoSchema):
         }
     )
 
+    @property
+    def is_bulk_action(self):
+        """Custom property for convenience."""
+        return hasattr(self.view, "action") and self.view.action in self.custom_actions
+
+    @property
+    def is_partial_action(self):
+        """Custom property for convenience."""
+        return hasattr(self.view, "action") and self.view.action in ["partial_update", "bulk_partial_update"]
+
+    def _get_paginator(self):
+        """Nautobot's custom bulk operations, even though they return a list of records, are NOT paginated."""
+        if self.is_bulk_action:
+            return None
+        return super()._get_paginator()
+
+    def get_filter_backends(self):
+        """Nautobot's custom bulk operations, even though they return a list of records, are NOT filterable."""
+        if self.is_bulk_action:
+            return []
+        return super().get_filter_backends()
+
+    def get_operation(self, *args, **kwargs):
+        operation = super().get_operation(*args, **kwargs)
+        # drf-spectacular never generates a requestBody for DELETE operations, but our bulk-delete operations need one
+        if "requestBody" not in operation and self.is_bulk_action and self.method == "DELETE":
+            # based on drf-spectacular's `_get_request_body()`, `_get_request_for_media_type()`,
+            # `_unwrap_list_serializer()`, and `_get_request_for_media_type()` methods
+            request_serializer = self.get_request_serializer()
+
+            # We skip past a number of checks from the aforementioned private methods, as this is a very specific case
+            component = self.resolve_serializer(request_serializer.child, "request")
+
+            operation["requestBody"] = {
+                "content": {
+                    media_type: build_media_type_object(build_array_type(component.ref))
+                    for media_type in self.map_parsers()
+                },
+                "required": True,
+            }
+
+        return operation
```

Comment:
Another ripe opportunity for a "v2 TODO" I hope? Such as if we can align around normalized responses? Or is the lack of filtering/pagination for these bulk responses always going to be an issue here? (Such as how GET will always be paginated, but all the others won't?)

Reply:
This part is specifically for bulk-DELETE; I think we could possibly push this upstream to get drf-spectacular to remove its code that hard-codes all DELETE request bodies to null.

Reply:
> This part is specifically for bulk-DELETE; I think we could possibly push this upstream to get drf-spectacular to remove its code that hard-codes all DELETE request bodies to null.

I like this idea.

Reply:
Opened https://github.com/tfranzel/drf-spectacular/issues/809.